src/darksend.cpp

```diff
@@ -2009,22 +2032,26 @@ bool CDarksendPool::IsOutputsCompatibleWithSessionDenom(const std::vector<CTxDSO
     return true;
 }
 
-bool CDarksendPool::IsDenomCompatibleWithSession(int nDenom, CTransaction txCollateral, int& nErrorID)
+bool CDarksendPool::IsDenomCompatibleWithSession(int nDenom, CTransaction txCollateral, PoolMessage& nMessageIDRet)
 {
-    if(nDenom == 0) return false;
+    if(nDenom == 0) {
+        nMessageIDRet = ERR_DENOM;
+        return false;
+    }
 
     LogPrintf("CDarksendPool::IsDenomCompatibleWithSession -- nSessionDenom: %d nSessionUsers: %d\n", nSessionDenom, nSessionUsers);
 
     if(!fUnitTest && !IsCollateralValid(txCollateral)) {
         LogPrint("privatesend", "CDarksendPool::IsDenomCompatibleWithSession -- collateral not valid!\n");
-        nErrorID = ERR_INVALID_COLLATERAL;
+        nMessageIDRet = ERR_INVALID_COLLATERAL;
         return false;
     }
 
     if(nSessionUsers < 0) nSessionUsers = 0;
 
     if(nSessionUsers == 0) {
-        nSessionID = 1 + GetRand(999999);
+        nMessageIDRet = MSG_NOERR;
+        nSessionID = GetInsecureRand(999999)+1;
```

Comment:
Are you sure replacing GetRand with GetInsecureRand is a good idea here ?  I don't know enough about the PS implementation to know how this gets used but it seems like a change that has potential security implications.

Also GetInsecureRand should really never be used.  As I've said before it isn't thread-safe due to it's use of global variables.  There's no guarantee it will always work correctly in a multithreaded program.